Foundation/FileManager.swift

```diff
@@ -1033,16 +1033,39 @@ open class FileManager : NSObject {
         NSUnimplemented()
     }
     
-    internal func _tryToResolveTrailingSymlinkInPath(_ path: String) -> String? {
+    internal func _tryToResolveTrailingSymlinkInPath(_ path: String, recursionLevel: Int = 0, initialPath: String? = nil) -> String? {
         guard _pathIsSymbolicLink(path) else {
             return nil
         }
         
```

Comment:
The call to `_pathIsSymbolicLink()` can be removed as `destinationOfSymbolicLink()` should fail anyway if the path is not a symbolic link (according to `readlink(2)`).

Reply:
The function `_pathIsSymbolicLink()` was already there so I decided not to touch it. It was introduced as is with the [initial `URL.resolvingSymlinksInPath()` implementation](https://github.com/apple/swift-corelibs-foundation/commit/df1b258).